prealloc.go

```diff
@@ -304,16 +318,8 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 		return v
 
 	}
-	// if simple is true, then we actually have to check if we had returns inside of our loop. Otherwise, we can just report all messages.
-	if v.simple == true && !v.returnsInsideOfLoop {
```

Comment:
It looks like we've lost our processing of `v.simple`? On a side note, thanks for taking on this issue! 👍